### PR TITLE
Scripts: Add RTLCSS to wp-scripts.

### LINKED ISSUE #29605: Add RTLCSS to wp-scripts
## What problem does this address?

`@wordpress/scripts` does not output RTL stylesheets.

## What is your proposed solution?

`@wordpress/scripts` should build RTL stylesheets out of the box.

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #29605 — This simply adds an RTLCSS step to the webpack build process in wp-scripts.

## Why?

Core supports switching out the CSS files automatically if a `[name]-rtl.css` exists, but wp-scripts doesn't generate that currently. This adds the rtlcss plugin, which automatically generates the RTL-ified file for any CSS files. These are then automatically loaded on RTL sites when registered in `block.json` (since `register_block_style_handle` [handles this](https://github.com/WordPress/wordpress-develop/blob/9c392836b4ed3f6f469b51218e1d11eea652a263/src/wp-includes/blocks.php#L342-L356)).

## How?

Adds [the RTLCSS plugin](https://rtlcss.com/).

## Testing Instructions

1. Using any project that would build with wp-scripts (for example, spin up a create-block project)
2. Run the bin/wp-scripts.js directly: `/path-to/gutenberg/packages/scripts/bin/wp-scripts.js build`
3. It should generate `-rtl.css` files

Note: I also tested this with a project that already includes the custom webpack code mentioned [in this comment](https://github.com/WordPress/gutenberg/issues/29605#issuecomment-796772630), and there were no issues with there being two RTLCSS plugins.

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
It works like a charm in my testing:

Screenshot 2024-05-10 at 10 09 56

File: packages/scripts/package.json [hunk 1]
```diff
@@ -79,6 +79,7 @@
 		"react-refresh": "^0.14.0",
 		"read-pkg-up": "^7.0.1",
 		"resolve-bin": "^0.4.0",
+		"rtlcss-webpack-plugin": "4.0.7",
```
Comment (gziolo) [inline packages/scripts/package.json]: Let's ensure we don't force people to pin this particular version:
```suggestion
		"rtlcss-webpack-plugin": "^4.0.7",
```

**Comment by swissspidy** [comment]
Wooohooo finally! 🎉

**Comment by ryelle** [comment]
🎉  Thanks for merging @gziolo!

**Comment by bacoords** [comment]
Thanks for this! Is there any option to opt-out of it, especially in instances where developers are extending `@wordpress/scripts` webpack config to use it for non-block stuff like editor modifications?

**Comment by swissspidy** [comment]
Array.prototype.filter is your friend, you can use it to exclude certain plugins from the default list.  Or just provide your own list.

**Comment by justintadlock** [comment]
Here's how I was able to exclude RTL in my webpack config:

```js
const defaultConfig = require( '@wordpress/scripts/config/webpack.config' );
const RtlCssPlugin  = require( 'rtlcss-webpack-plugin' );

module.exports = {
	...defaultConfig,
	...{
		plugins: [
			...defaultConfig.plugins.filter(
				(filter) => ! (filter instanceof RtlCssPlugin)
			),
			// Other plugins.
		]
	}
};
```

**Comment by KevinBatdorf** [comment]
The above stopped working for me but this works to remove it:
```js
...defaultConfig.plugins.filter(
  (filter) =>
    // Remove the rtl plugin
    !(filter instanceof RtlCssPlugin) &&
      filter.constructor.name !== 'RtlCssPlugin',
),
```
https://github.com/KevinBatdorf/code-block-pro/blob/main/webpack.config.js#L14

**Comment by justintadlock** [comment]
Note: As of https://github.com/WordPress/gutenberg/pull/68201, Gutenberg switched to a fork of the RTL CSS plugin, so referencing it in `webpack.config.js` has changed:

```js
const RtlCssPlugin = require('@wordpress/scripts/plugins/rtlcss-webpack-plugin');
```

**Pull request merged**